projects/plugins/boost/tests/e2e/specs/jetpack-boost/critical-css.test.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * External dependencies
+ */
+import { prerequisitesBuilder } from 'jetpack-e2e-tests/lib/env/prerequisites';
+
+/**
+ * Internal dependencies
+ */
+import JetpackBoostPage from '../../lib/pages/wp-admin/JetpackBoostPage';
+
+// TODO: This is for illustrative purpose only. It will need refactoring and improving.
+describe( 'Critical CSS module', () => {
+	let jetpackBoostPage;
+	beforeAll( async () => {
+		await prerequisitesBuilder().withLoggedIn( true ).withConnection( true ).build();
+	} );
+
+	beforeEach( async function () {
+		jetpackBoostPage = await JetpackBoostPage.visit( page );
+	} );
+
+	afterEach( async function () {
+		if ( jetpackBoostPage ) {
+			await jetpackBoostPage.page.close();
+		}
+	} );
+
+	it( 'should allow enabling critical css', async () => {
+		const toggle = await page.$( '#jb-feature-toggle-critical-css' );
```

Comment:
It's preferred to have the element location and interaction inside the page object. So, instead you'll call something like `page.toggleOptimizeCss()`.
A rule that helps it not to have element selectors in tests, only in pages.

Looking at this test, I would see it repeatingly using 2 functions from JetpackBoostPage:
```
page.toggleFeatureX()
page.isFeatureXActive()
```

That should make the test much cleaner and much easier to read and understand.

Reply:
Thanks. This is very useful feedback. I have now refactored this in commit 5999a6ceed69df20fafff8fa227b4fa4603d62b3. Then it allowed me to easily add more tests for other modules in commit 5999a6ceed69df20fafff8fa227b4fa4603d62b3 and 5999a6ceed69df20fafff8fa227b4fa4603d62b3

I also refactored the Speed Score feature tests too in commit ae9cfee6aab557fd4ef78092a7971ceb44f312be